// angular2-qrcode/lib/angular2-qrcode.umd.js
(function (global, factory) {
	typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/core'), require('qrious')) :
	typeof define === 'function' && define.amd ? define(['exports', '@angular/core', 'qrious'], factory) :
	(factory((global['angular2-qrcode'] = global['angular2-qrcode'] || {}),global._angular_core,global.QRious));
}(this, (function (exports,_angular_core,QRious) { 'use strict';

QRious = 'default' in QRious ? QRious['default'] : QRious;

var QRCodeComponent = (function () {
    /**
     * @param {?} elementRef
     */
    function QRCodeComponent(elementRef) {
        this.elementRef = elementRef;
        this.background = 'white';
        this.backgroundAlpha = 1.0;
        this.foreground = 'black';
        this.foregroundAlpha = 1.0;
        this.level = 'L';
        this.mime = 'image/png';
        this.padding = null;
        this.size = 100;
        this.value = '';
        this.canvas = false;
    }
    /**
     * @param {?} changes
     * @return {?}
     */
    QRCodeComponent.prototype.ngOnChanges = function (changes) {
        if ('background' in changes ||
            'backgroundAlpha' in changes ||
            'foreground' in changes ||
            'foregroundAlpha' in changes ||
            'level' in changes ||
            'mime' in changes ||
            'padding' in changes ||
            'size' in changes ||
            'value' in changes ||
            'canvas' in changes) {
            this.generate();
        }
    };
    /**
     * @return {?}
     */
    QRCodeComponent.prototype.generate = function () {
        try {
            var /** @type {?} */ el = this.elementRef.nativeElement;
            el.innerHTML = '';
            var /** @type {?} */ qr = new QRious({
                background: this.background,
                backgroundAlpha: this.backgroundAlpha,
                foreground: this.foreground,
                foregroundAlpha: this.foregroundAlpha,
                level: this.level,
                mime: this.mime,
                padding: this.padding,
                size: this.size,
                value: this.value
            });
            if (this.canvas) {
                el.appendChild(qr.canvas);
            }
            else {
                el.appendChild(qr.image);
            }
        }
        catch (e) {
            console.error("Could not generate QR Code: " + e.message);
        }
    };
    return QRCodeComponent;
}());
QRCodeComponent.decorators = [
    { type: _angular_core.Component, args: [{
                moduleId: 'module.id',
                selector: 'qr-code',
                template: ""
            },] },
];
/**
 * @nocollapse
 */
QRCodeComponent.ctorParameters = function () { return [
    { type: _angular_core.ElementRef, },
]; };
QRCodeComponent.propDecorators = {
    'background': [{ type: _angular_core.Input },],
    'backgroundAlpha': [{ type: _angular_core.Input },],
    'foreground': [{ type: _angular_core.Input },],
    'foregroundAlpha': [{ type: _angular_core.Input },],
    'level': [{ type: _angular_core.Input },],
    'mime': [{ type: _angular_core.Input },],
    'padding': [{ type: _angular_core.Input },],
    'size': [{ type: _angular_core.Input },],
    'value': [{ type: _angular_core.Input },],
    'canvas': [{ type: _angular_core.Input },],
};
var QRCodeModule = (function () {
    function QRCodeModule() {
    }
    return QRCodeModule;
}());
QRCodeModule.decorators = [
    { type: _angular_core.NgModule, args: [{
                exports: [QRCodeComponent],
                declarations: [QRCodeComponent]
            },] },
];
/**
 * @nocollapse
 */
QRCodeModule.ctorParameters = function () { return []; };

/**
 * Generated bundle index. Do not edit.
 */

exports.QRCodeComponent = QRCodeComponent;
exports.QRCodeModule = QRCodeModule;

Object.defineProperty(exports, '__esModule', { value: true });

})));
